### PR TITLE
v2.3.0

### PR DESCRIPTION
**Pull request by notgull**
- Add `StreamExt::try_iter` for polling a `Stream` while it is ready. (#70)

### CODE REVIEW

**Comment by notgull** [comment]
Build failure is #95

**Comment by taiki-e** [comment]
- Adding a new API should bump the minor version.
- The actual API added does not appear to be try_iter.
  https://github.com/smol-rs/futures-lite/pull/70/files#diff-aa0a8d713ea64389a4b4916384d286b01c0de291edf10fec2f213123419e2647R1813
- And that name seems to be used in a different sense than the [common try-stream in the ecosystem](https://docs.rs/futures/latest/futures/stream/trait.TryStream.html)?

**Comment by notgull** [comment]
> * And that name seems to be used in a different sense than the [try-stream common in the ecosystem](https://docs.rs/futures/latest/futures/stream/trait.TryStream.html)?

Hmm, good point. What if we called it `ready_for(&stream)` instead?

**Comment by taiki-e** [comment]
> ready_for

Hmm, "ready" is already used in multiple meanings... (task::ready!, future::ready)

**Comment by notgull** [comment]
Scattershot other suggestions:

- `stream_iter`
- `immediate`
- `poll_once`

**Comment by i509VCB** [comment]
`drain` might work quite well.

Drain would also match the semantics around [Vec::drain](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain) which is an iterator that also removes every element from the container.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Version 2.3.0
+
+- Add `Stream:drain` for draining objects from a `Stream` without waiting (#70).
```
Comment (taiki-e) [inline CHANGELOG.md:3]:
```suggestion
- Add `StreamExt:drain` for draining objects from a `Stream` without waiting (#70).
```

**Review by taiki-e — APPROVED** [review]

**Pull request merged**